### DISCUSSION TITLE
init app without async

### DISCUSSION THREAD

**Discussion by antn9x** [pixijs/pixijs#10943] [v8 General]
How can I init app without async/await? 
`await app.init({});`
i want to remove `await`.

**Comment by bigtimebuddy** [comment]
You cannot init synchronously, but you can use the promise-then syntax instead:

```js
app.init({}).then(() => {
  // done initializing!
});
```

**Reply by antn9x** [reply]
No, i want to sync approach, not `then`

**Reply by bigtimebuddy** [reply]
Then use v7. v8, which supports WebGPU, requires this step to be async.

**Reply by antn9x** [reply]
yes, i used v7, is there anyway to select renderer to ignore async?

**Reply by bigtimebuddy** [reply]
No, there isn't. Why do you need this?

**Reply by antn9x** [reply]
I have some issue with HMR, but passed with setTimeout. async function is problem here.

**Comment by shevernitskiy** [comment]
Just in case, if you want it sync because you of usage inside constructor, you can make a class with static async factory and make constructor private

**Reply by antn9x** [reply]
can you provide some example code?

**Reply by shevernitskiy** [reply]
Something like
```ts
class MyPainter {
  app: Application

  private constructor(params: Params) {}

  static async create(params: Params) {
    const instance = new MyPainter(params);
    instance.app = new Application();
    await instance.app.init();
    return instance;
  }
}

const painter = await MyPainter.create(params);
```
